Change g_indent into ntabs parameter of Dump(unsigned int ntabs)

LGM-Doyle opened the pull request:
#### Problem ####
Multiple threads increment/decrement `g_indent` without any synchonization.  

I think that this leads to rare buggy behavior every few thousand turns.  It causes data corruption, log corruption and/or crashes.

#### This Solution ####
Convert `g_indent` from a global that requires synchronization to a parameter of `Dump()`

LGM-Doyle committed:
Change g_indent into ntabs parameter of Dump(unsigned int ntabs)

g_indent was used from multiple threads without any thread
synchronization.

dbenage-cx reviewed (approved):
LGTM, consider comment optional.

LGM-Doyle committed:
Improve code clarity: Don't repeatedly increment then decrement ntabs.

As noted in review #1917, repeatedly incrementing and decrementing ntabs
before calling DumpIndent(ntabs) is less clear than simply indicating
the tab level with a +1 or +2.
